### DISCUSSION TITLE
unit conversion using the API

### DISCUSSION THREAD

**Discussion by theedge456** [Qalculate/libqalculate#733] [Q&A] [answered]
Hello,
I'm testing the API to do a basic conversion.
 It seems not the correct way to proceed.
Any hint ?
```
 Calculator c;
 c.loadGlobalUnits();
Unit* uFrom=c.getUnit("meter");
 Unit* uTo=c.getCompositeUnit("r:mm_c"); // returns NULL
MathStructure ms = c.convert(10,uFrom, uTo);
```

**Comment by hanna-kn** [comment]
Remove `r:` from `r:mm_c` (letters before colon in data files specifies properties of name/symbol).

**Reply by hanna-kn** [reply]
A simpler and more flexible alternative to using getCombositeUnit() is to create a temporary unit, e.g. `CompositeUnit uTo("", "", "", "mm")`

Other alternatives include `c.convert("10m", "mm")` and `c.calculate("10m->mm")` (and related functions).

**Reply by theedge456** [reply]
Thanks for the hints.
But none seems to give the results.
```
MathStructure ms2=c.calculate("10m->mm");
PrintOptions po2;
std::cout<< ms2.print(po2); // shows: 10 m^-1 m*m
MathStructure ms1=c.convert("10m", "mm");
      PrintOptions po1;
      ms1.format(po1);
      std::cout<< ms1.print(po1); //  shows: 10 m
```
What is missing ?
Are there samples of code using the API somewhere ?

**Reply by hanna-kn** [reply, marked as answer]
Prefixes has not been loaded:
`c.loadGlobalPrefixes();`

`loadGlobalDefinitions()` is however preferable (a few units depend on variables).

`c.convert("10m", "mm")` should be `c.convert(c.parse("10m"), "mm")`

> Are there samples of code using the API somewhere ?

No (not that I know of, except the code of the user interfaces). I generally recommend use of `Calculator::calculateAndPrint()` for simple input and output.

**Reply by theedge456** [reply]
Thanks for the tips.
